debezium-core/src/main/java/io/debezium/spatial/GeometryEndiannessConverter.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.spatial;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.Arrays;
+
+import io.debezium.DebeziumException;
+
+/**
+ * Converts a geometry's byte order (endianness).
+ *
+ * @author Chris Cranford
+ */
+class GeometryEndiannessConverter {
+
+    /**
+     * Creates a new {@link GeometryBytes} using little endian format.
+     *
+     * @param geometry the geometry to convert, must not be {@code null}
+     * @return the converted geometry
+     */
+    public static GeometryBytes toLittleEndian(GeometryBytes geometry) {
+        return convert(geometry, ByteOrder.LITTLE_ENDIAN);
+    }
+
+    /**
+     * Creates a new {@link GeometryBytes} using big endian format.
+     *
+     * @param geometry the geometry to convert, must not be {@code null}
+     * @return the converted geometry
+     */
+    public static GeometryBytes toBigEndian(GeometryBytes geometry) {
+        return convert(geometry, ByteOrder.BIG_ENDIAN);
+    }
+
+    private static GeometryBytes convert(GeometryBytes geometry, ByteOrder targetByteOrder) {
+        try {
```

Comment:
Maybe here could be a check if the byte order of the source data is the same as target byte order then no operation is needed?